### PR TITLE
Update PowerShell Command.

### PR DESCRIPTION
**Pull request by RandomNoun7**
Update the powershell command to work with the rename of the executable to pwsh on non windows platforms.

### CODE REVIEW

**Comment by RandomNoun7** [comment]
I see now the indentation doesn't match original. I can fix and resubmit if you like.

**Comment by michaeltlombardi** [comment]
Is there a reliable way to discover if `pwsh` is available on Windows?

**Comment by RandomNoun7** [comment]
IF BLOCK ALL THE THINGS! Updating now.

**Comment by RandomNoun7** [comment]
Added if blocks to test for pwsh first and then fallback to powershell.exe, even on Windows.

File: lib/puppet/provider/package/psmodule.rb [hunk 1]
```diff
@@ -11,13 +11,19 @@
...
-                "#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\powershell.exe"
-              elsif File.exist?("#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe")
-                "#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe"
-              else
-                'powershell.exe'
-              end
+    if File.exists?("#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\powershell.exe")
+      "#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\pwsh.exe"
+    elsif File.exists?("#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe")
+      "#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\pwsh.exe"
```
Comment (michaeltlombardi) [inline lib/puppet/provider/package/psmodule.rb]: These four lines seem to look for powershell and then specify pwsh?

File: lib/puppet/provider/psrepository/windows.rb [hunk 2]
```diff
@@ -3,13 +3,19 @@
...
-                "#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\powershell.exe"
-              elsif File.exist?("#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe")
-                "#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe"
-              else
-                'powershell.exe'
-              end
+    if File.exists?("#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\powershell.exe")
+      "#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\pwsh.exe"
+    elsif File.exists?("#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe")
+      "#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\pwsh.exe"
```
Comment (michaeltlombardi) [inline lib/puppet/provider/psrepository/windows.rb]: As above

**Comment by hbuckle** [comment]
Hi, thanks for the PR and comments. For Linux just using pwsh should be fine but on Windows PowerShell 6 installs alongside 5, and they use different modulepaths so I need to work out how best to handle that. I'll do a bit of experimenting tomorrow.

**Comment by michaeltlombardi** [comment]
@hbuckle - looking into the module path stuff, it turns out that the module paths aren't shared between versions at all.

https://github.com/PowerShell/PowerShell/issues/4599

Every version has it's own cache of modules. My instinct would be to initially consider this works-as-intended and not try to reconcile across versions, especially if the forthcoming fix is going to live in pwsh `6.1` or something.

**Comment by hbuckle** [comment]
I've now published v1.0.0 which supports pwsh and powershell by having a separate provider for each. Sorry it took so long, there was an issue on Linux with the Install-Module command failing which took me a while to figure out.

**Pull request closed without merging**